frontend/src/context/CartContext.tsx

```diff
@@ -0,0 +1,184 @@
+import { createContext, useContext, useState, useEffect, ReactNode } from 'react';
+import axios from 'axios';
+import { api } from '../api/config';
+
+interface Product {
+  productId: number;
+  name: string;
+  description: string;
+  price: number;
+  imgName: string;
+  sku: string;
+  unit: string;
+  supplierId: number;
+  discount?: number;
+}
+
+interface CartItem {
+  cartItemId: number;
+  cartId: number;
+  productId: number;
+  quantity: number;
+  addedAt: string;
+  product: Product;
+}
+
+interface Cart {
+  cartId: number;
+  sessionId: string;
+  createdAt: string;
+  updatedAt: string;
+  items: CartItem[];
+}
+
+interface CartContextType {
+  cart: Cart | null;
+  loading: boolean;
+  addToCart: (productId: number, quantity: number) => Promise<void>;
+  updateQuantity: (cartItemId: number, quantity: number) => Promise<void>;
+  removeItem: (cartItemId: number) => Promise<void>;
+  clearCart: () => Promise<void>;
+  refreshCart: () => Promise<void>;
+  getTotalItems: () => number;
+  getTotalPrice: () => number;
+}
+
+const CartContext = createContext<CartContextType | undefined>(undefined);
+
+const CART_SESSION_KEY = 'octocat_cart_session';
+
+const getSessionId = (): string => {
+  let sessionId = localStorage.getItem(CART_SESSION_KEY);
+  if (!sessionId) {
+    sessionId = `session_${Date.now()}_${Math.random().toString(36).substring(2, 15)}`;
+    localStorage.setItem(CART_SESSION_KEY, sessionId);
+  }
+  return sessionId;
+};
+
+export function CartProvider({ children }: { children: ReactNode }) {
+  const [cart, setCart] = useState<Cart | null>(null);
+  const [loading, setLoading] = useState(true);
+  const sessionId = getSessionId();
+
+  const refreshCart = async () => {
+    try {
+      setLoading(true);
+      const { data } = await axios.get(`${api.baseURL}/api/cart/${sessionId}`);
+      // API returns {cart, items}, transform to expected structure
+      setCart({
+        ...data.cart,
+        items: data.items || [],
+      });
+    } catch (error) {
+      console.error('Error fetching cart:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const loadCart = async () => {
+      try {
+        setLoading(true);
+        const { data } = await axios.get(`${api.baseURL}/api/cart/${sessionId}`);
+        // API returns {cart, items}, transform to expected structure
+        setCart({
+          ...data.cart,
+          items: data.items || [],
+        });
+      } catch (error) {
+        console.error('Error fetching cart:', error);
+      } finally {
+        setLoading(false);
+      }
+    };
+    loadCart();
```

Comment:
The cart loading logic is duplicated between `refreshCart` (lines 64-78) and the `useEffect` callback (lines 81-95). Both perform identical operations: set loading, fetch cart, transform data, handle errors. Consider extracting this into a single function or having the useEffect simply call `refreshCart()` to eliminate duplication and ensure consistent behavior.
```suggestion
    refreshCart();
```